frontend/src/components/Navbar.js

```diff
@@ -12,12 +13,58 @@ import MenuItem from "@material-ui/core/MenuItem"
 import AccountCircle from "@material-ui/icons/AccountCircle"
 import NotificationsIcon from "@material-ui/icons/Notifications"
 import Typography from "@material-ui/core/Typography"
+import MenuIcon from "@material-ui/icons/Menu"
 import GsTitle from "./GsTitle"
+import clsx from "clsx"
+
+const NavHeader = observer(({ drawerOpen, handleDrawerOpen, drawerWidth }) => {
```

Comment:
I moved the drawer into the Navbar component to fix some of the header nesting issues. Needed to pass some props down for this.